autogen/generator.py

```diff
@@ -320,11 +320,16 @@ def __call__(self):
         D = sorted(D.values(), key=lambda d: d['function'])
         sys.stdout.write("Generating PARI functions:")
 
-        self.gen_file = open(self.gen_filename + '.tmp', 'w')
+        # Stupid Python 3 forces us to specify an encoding
+        if "encoding" in open.__doc__:
+            kwds = dict(mode="wt", encoding="utf-8")
+        else:
+            kwds = dict(mode="wt")
```

Comment:
One thing we really should have proposed for Python when trying to improve the 2/3 compatibility situation is to add a doctest directive to ignore unicode literal markers, b.c. I've had to reimplement that multiple times...  I think in general doctests just aren't used as extensively in much of the Python community, but still.

Also I don't think it's necessary to put `unicode_literals` absolutely everywhere but I haven't done a careful analysis of where the bytes/unicode boundaries are in the autogen code :/